toolkit/offchain/src/csl.rs

```diff
@@ -165,6 +169,26 @@ impl OgmiosUtxoExt for OgmiosUtxo {
 	fn to_csl(&self) -> Result<TransactionUnspentOutput, JsError> {
 		Ok(TransactionUnspentOutput::new(&self.to_csl_tx_input(), &self.to_csl_tx_output()?))
 	}
+
+	fn to_constructor_style_csl_plutus_data(&self) -> PlutusData {
+		PlutusData::new_constr_plutus_data(&cardano_serialization_lib::ConstrPlutusData::new(
+			&0u64.into(),
+			&{
+				let mut list = cardano_serialization_lib::PlutusList::new();
+				list.add(&PlutusData::new_single_value_constr_plutus_data(
+					&0u64.into(),
+					&PlutusData::new_bytes(self.transaction.id.to_vec()),
+				));
+				list.add(&PlutusData::new_integer(&(self.index as u64).into()));
+				list
+			},
+		))
```

Comment:
We have this logic in `to_datum` of domain type. I would use it here instead of duplicating.